app/src/main/java/com/example/feelgoodinc/services/UserService.java

```diff
@@ -125,14 +123,19 @@ public void loginUser(String email, String password, UserCallback callback){
      * @param callback interface to track success/error and send data back to activity
      */
     public void registerUser(String email, String password, UserCallback callback){
-        mAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(task -> {
-            if (task.isSuccessful()) {
-                User user = getCurrentUser();
-                callback.onSuccess(user);
-            } else {
-                callback.onError(task.getException());
-            }
-        });
+        if (validatePassword(password)){
+            mAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(task -> {
+                if (task.isSuccessful()) {
+                    User user = getCurrentUser();
+                    callback.onSuccess(user);
+                } else {
+                    callback.onError(task.getException());
+                }
+            });
+        }
+
+        // give a specific error if the password is invalid
+        callback.onError(new IllegalArgumentException("Password does not meet the required criteria."));
```

Comment:
Wouldn't this line get triggered 100% of the time, regardless of if the code goes through the if or not?